### PR TITLE
Remove concurrent_layout and clarify layout and themes selections

### PR DESCRIPTION
**Pull request by bradymiller**
First pass to remove concurrent_layout and clarify layout and themes selections.

(note I added below explanation to this entry on 9/25/16 1:47 PST)
Goals of this commit:

First goal is to simplify the selections in globals since it is now confusing. This commit changes it to the following options:
1. 'Layout' with options of 'Frames' or 'Tabs'
2. 'Tabs Layout Theme' with options of 'Full', 'Compact' and any other tabs themes that are added on later
3. 'General Theme' with the standard options for themes
4. 'Vertical Menu Style' with options of 'Tree' and 'Sliding'

Second goal is to remove the concurrent_layout global since it is no longer needed (and just functions as a source of confusion for developers) and remove all the code related to no frames, radio buttons, and tree menu view settings. Thus if choose Frames above, will then see the sliding menu. If we need to keep the tree menu view option then would just create another global for this that is specific to the menu since the coding to do that part (differentiating from tree menu and sliding menu) is very minimal.

### CODE REVIEW

File: interface/forms_admin/forms_admin.php [hunk 1]
```diff
@@ -26,7 +26,7 @@
 }
 $bigdata = getRegistered("%") or $bigdata = false;
 
-$formtarget = $GLOBALS['concurrent_layout'] ? "" : " target='Main'";
+$formtarget = "";
```
Comment (bradymiller) [inline interface/forms_admin/forms_admin.php]: dig deeper and removed this variable from script

File: interface/globals.php [hunk 2]
```diff
@@ -269,6 +269,10 @@ function undoMagicQuotes($array, $topLevel=true) {
     $GLOBALS['language_menu_login'] = true;
   }
 
+  // Add this $GLOBALS['concurrent_layout'] == 3 in order to support legacy forms
+  // that may use this. Note this global is no longer uses in the standard codebase
```
Comment (bradymiller) [inline interface/globals.php]: Fix grammar issues

File: interface/main/left_nav.php [hunk 3]
```diff
@@ -160,9 +160,6 @@
   'dld' => array(xl('Display Documents'), 0, 'main/display_documents.php')
  );
  $primary_docs['npa']=array(xl('Batch Payments')   , 0, 'billing/new_payment.php');
- if ($GLOBALS['use_charges_panel'] || $GLOBALS['concurrent_layout'] == 2) {
-  $primary_docs['cod'] = array(xl('Charges'), 2, 'patient_file/encounter/encounter_bottom.php');
- }
```
Comment (bradymiller) [inline interface/main/left_nav.php]: Oops. Add this back and just remove the concurrent_layout part.

File: interface/main/left_nav.php [hunk 4]
```diff
@@ -1474,50 +1412,6 @@ function selpopup(selobj) {
...
-  if ($key == 'msg') echo " checked";
-  echo " /></td>\n";
-  echo " </tr>\n";
- }
-?>
-</table>
-
-<?php } ?>
-
 <br /><hr />
```
Comment (bradymiller) [inline interface/main/left_nav.php]: see if can remove js functions that are no longer used

File: interface/patient_file/encounter/diagnosis_full.php [hunk 5]
```diff
@@ -3,7 +3,7 @@
 require_once("$srcdir/billing.inc");
 require_once("$srcdir/formdata.inc.php");
 
-$targparm = $GLOBALS['concurrent_layout'] ? "" : "target='Main'";
+$targparm = "";
```
Comment (bradymiller) [inline interface/patient_file/encounter/diagnosis_full.php]: dig deeper and remove the variable

File: library/globals.inc.php [hunk 6]
```diff
@@ -136,13 +135,13 @@
...
     'new_tabs_layout' => array(
-      xl('Tabs Layout (need to logout/login after change this setting)'),
+      xl('Layout (need to logout/login after change this setting)'),
       array(
-        '0' => xl('Off'),
-        '1' => xl('On'),
+        '0' => xl('Frame'),
+        '1' => xl('Tabs'),
       ),
       '1',                              // default = off
```
Comment (bradymiller) [inline library/globals.inc.php]: change comment to default = 1

**Comment by sunsetsystems** [comment]
Not sure I understand where you are going with this commit.  Are you doing away with the tree menu or sliding menu?  concurrent_layout is used to distinguish between those.

**Comment by bradymiller** [comment]
Sorry, I should of provided a better description of this code.
Goals of this commit:

First goal is to simplify the selections in globals since it is now confusing. This commit changes it to the following options:
1. 'Layout' with options of 'Frames' or 'Tabs'
2. 'Tabs Layout Theme' with  options of 'Full', 'Compact' and any other tabs themes that are added on later
3. 'General Theme' with the standard options for themes

Second goal is to remove the concurrent_layout global since it is no longer needed (and just functions as a source of confusion for developers) and remove all the code related to no frames, radio buttons, and tree menu view settings. Thus if choose Frames above, will then see the sliding menu. If we need to keep the tree menu view option then would just create another global for this that is specific to the menu since the coding to do that part (differentiating from tree menu and sliding menu) is very minimal.

thoughts?

**Comment by sunsetsystems** [comment]
I agree with removing the original old-style layout and the radio buttons menu style (that can simplify left_nav quite a bit).  However I think we should keep both the sliding menu and the tree menu options because they are mostly based on the same code (main difference is CSS as I recall) and the tree menu is quite nice.  Towards that end it seems right to have a single dropdown to select one of the 3 layout types.

Alternatively it might be appealing to completely separate the menu style choice from the tabbed layout choice.  I was wondering why we couldn't have a tabbed layout with one of the existing menu styles on the left instead of on top.

**Comment by robertdown** [comment]
I really disagree with mixing the tabs and left style menu. It’s a great time to ditch the radio button menu and original style. I think the options should be sliding, tree menu, and tabs. I’d like to see the eventual drop of the left nav style menu all together

**Comment by sunsetsystems** [comment]
Hi Robert, what's your objection to the left menu?  Given that most screens are very wide now, I have found menus on the left or right to be a more practical use of space.  However I'm not a clinician so there's probably something I'm missing.

**Comment by robertdown** [comment]
The objection is mostly the poor code behind the left menu. I agree that we've got screens that are plenty wide now, but from a practical point, the underlying code that powers the left frame is basically at a point where I won't even touch it. Moving to the tabs based is a good opportunity to bring the code standards up to par, something I think is invaluable and has been missing from OpeEMR for too long

**Comment by sunsetsystems** [comment]
Well, code quality and style is a different question.  I was talking about the concept of left vs. top and notion of having a menu style that's divorced from the choices for tabs and frames.  And was mostly food for thought, not something to do now.

File: interface/main/left_nav.php (hunk 3 above)
Comment (bradymiller) [inline interface/main/left_nav.php]: Add a menu_styling_vertical global which does:
0 for Tree
1 for Sliding
(and incorportate if this global == 0 above in above if)

File: interface/main/left_nav.php [hunk 7]
```diff
@@ -591,34 +576,11 @@ function syncRadios() {
     if (active_encounter == 0 && usage > '1') da = true;
     if (encounter_locked && usage > '1') da = true;
     <?php
-    if ($GLOBALS['concurrent_layout'] == 2){
```
Comment (bradymiller) [inline interface/main/left_nav.php]: put this back with menu_styling_vertical == 0

File: interface/main/left_nav.php [hunk 8]
```diff
@@ -1032,24 +993,6 @@ function selpopup(selobj) {
         $(" > a", this).addClass("collapsed");
       }
     });
-  }else if(2 == <?php echo $GLOBALS['concurrent_layout'] ?>){
```
Comment (bradymiller) [inline interface/main/left_nav.php]: put this back with menu_styling_vertical == 0

File: interface/main/left_nav.php [hunk 9]
```diff
@@ -995,7 +957,6 @@ function selpopup(selobj) {
  }
 // Treeview activation stuff:
 $(document).ready(function(){
-  if(3 == <?php echo $GLOBALS['concurrent_layout'] ?>){
```
Comment (bradymiller) [inline interface/main/left_nav.php]: put this back with menu_styling_vertical == 1

File: interface/main/left_nav.php [hunk 10]
```diff
@@ -1083,11 +1025,7 @@ function selpopup(selobj) {
  </tr>
 </table>
 
-<?php if ( $GLOBALS['concurrent_layout'] == 3) { ?>
```
Comment (bradymiller) [inline interface/main/left_nav.php]: put this back with menu_styling_vertical == 1

File: interface/main/left_nav.php [hunk 11]
```diff
@@ -1083,11 +1025,7 @@ function selpopup(selobj) {
...
   <ul id="navigation-slide">
-<?php } else { // ($GLOBALS['concurrent_layout'] == 2) ?>
-  <ul id="navigation">
-<?php } ?>
```
Comment (bradymiller) [inline interface/main/left_nav.php]: put this back

File: library/globals.inc.php [hunk 12]
```diff
@@ -174,13 +168,6 @@
...
 
-    'css_header' => array(
-      xl('Theme'),
-      'css',
-      'style_light.css',
-      xl('Pick a CSS theme.')
-    ),
-
     'gbl_nav_area_width' => array(
       xl('Navigation Area Width'),
```
Comment (bradymiller) [inline library/globals.inc.php]: Add a menu_styling_vertical (Vertical Menu Style) global which does:
0 for Tree
1 for Sliding

File: interface/forms_admin/forms_admin.php (hunk 1 above)
Comment (bradymiller) [inline interface/forms_admin/forms_admin.php]: DONE

File: interface/globals.php (hunk 2 above)
Comment (bradymiller) [inline interface/globals.php]: DONE

File: interface/main/left_nav.php (hunk 3 above)
Comment (bradymiller) [inline interface/main/left_nav.php]: DONE

File: interface/main/left_nav.php (hunk 7 above)
Comment (bradymiller) [inline interface/main/left_nav.php]: DONE

File: interface/main/left_nav.php (hunk 8 above)
Comment (bradymiller) [inline interface/main/left_nav.php]: DONE

File: interface/main/left_nav.php (hunk 9 above)
Comment (bradymiller) [inline interface/main/left_nav.php]: DONE

File: interface/main/left_nav.php (hunk 10 above)
Comment (bradymiller) [inline interface/main/left_nav.php]: DONE

File: interface/main/left_nav.php (hunk 11 above)
Comment (bradymiller) [inline interface/main/left_nav.php]: DONE

File: interface/main/left_nav.php (hunk 4 above)
Comment (bradymiller) [inline interface/main/left_nav.php]: Removed the setRadio function and all related code.

Comment (bradymiller) [inline interface/main/left_nav.php]: DONE

File: interface/patient_file/encounter/diagnosis_full.php (hunk 5 above)
Comment (bradymiller) [inline interface/patient_file/encounter/diagnosis_full.php]: DONE

File: library/globals.inc.php (hunk 6 above)
Comment (bradymiller) [inline library/globals.inc.php]: DONE

File: library/globals.inc.php (hunk 12 above)
Comment (bradymiller) [inline library/globals.inc.php]: DONE

**Comment by bradymiller** [comment]
Made some changes and I think it makes more sense now. I have separated the global for the layout from the global for the vertical menu styling(since it is definitely possible to migrate the top menu in the tabs layout to be a vertical menu if desired). This code also removes the old layout and the radio layout completely along with the setRadio function and all calls to it. Here are the global selections:
1. 'Layout' with options of 'Frames' or 'Tabs'
2. 'Tabs Layout Theme' with options of 'Full', 'Compact' and any other tabs themes that are added on later
3. 'General Theme' with the standard options for themes
4. 'Vertical Menu Style' with options of 'Tree' and 'Sliding'

It's testing well so far also. Let me know if this strategy is good or not.
thanks,
-brady

**Comment by sunsetsystems** [comment]
Nice.  I have not tested but the approach looks good to me.

**Pull request merged**